src/renderer/index.tsx

```diff
@@ -4,93 +4,46 @@
 
 import 'renderer/css/index.css';
 
-import React from 'react';
+import React, {useEffect} from 'react';
 import ReactDOM from 'react-dom';
 
-import type {CombinedConfig} from 'types/config';
-
 import MainPage from './components/MainPage';
+import {useConfig} from './hooks/useConfig';
 import IntlProvider from './intl_provider';
+import {printVersion} from './utils';
 
-type State = {
-    config?: CombinedConfig;
-}
-
-class Root extends React.PureComponent<Record<string, never>, State> {
-    constructor(props: Record<string, never>) {
-        super(props);
-        this.state = {};
-    }
-
-    async componentDidMount() {
-        await this.setInitialConfig();
-
-        window.desktop.onSynchronizeConfig(() => {
-            this.reloadConfig();
-        });
+function Root() {
```

Comment:
There is one small difference while doing the migration. This happens also on other files you have migrated to function components.

Many of these components extended from PureComponent, which had memoization attached to them. In order to have the exact same thing on function components, you have to wrap the component into a `React.memo` call. For example...

```
function MyComponent() {...}
export default React.memo(MyComponent);
```

That being said, I don't think it is such a big deal.

Reply:
I'll leave for now. We should re-evaluate though once we go to clean-up the front-end.